frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import {
+  Button,
+  Checkbox,
+  Form,
+  FormGroup,
+  FormSection,
+  PageSection,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { OpenDrawerRightIcon } from '@patternfly/react-icons';
+import { ConnectionTypeField } from '~/concepts/connectionTypes/types';
+import NameDescriptionField from '~/concepts/k8s/NameDescriptionField';
+import ConnectionTypePreviewDrawer from '~/concepts/connectionTypes/ConnectionTypePreviewDrawer';
+import { translateDisplayNameForK8s } from '~/concepts/k8s/utils';
+import ApplicationsPage from '~/pages/ApplicationsPage';
+import { NameDescType } from '~/pages/projects/types';
+import { CreateConnectionTypeFooter } from './CreateConnectionTypeFooter';
+import { CreateConnectionTypeFieldsTable } from './CreateConnectionTypeFieldsTable';
+import { CreateConnectionTypeBreadcrumbs } from './CreateConnectionTypeBreadcrumbs';
+import { createConnectionTypeObj } from './CreateConnectionTypeUtils';
+
+type CreateConnectionTypePageProps = {
+  prefillNameDesc?: NameDescType;
+  prefillEnabled?: boolean;
+  prefillFields?: ConnectionTypeField[];
+};
+
+export const CreateConnectionTypePage: React.FC<CreateConnectionTypePageProps> = ({
+  prefillNameDesc,
+  prefillEnabled,
+  prefillFields,
+}) => {
+  const [isDrawerExpanded, setIsDrawerExpanded] = React.useState(false);
+
+  const [connectionNameDesc, setConnectionNameDesc] = React.useState<NameDescType>(
+    prefillNameDesc || {
+      name: '',
+      k8sName: undefined,
+      description: '',
+    },
+  );
+  const [connectionEnabled, setConnectionEnabled] = React.useState<boolean>(
+    prefillEnabled || false,
+  );
+  const [connectionFields] = React.useState<ConnectionTypeField[]>(prefillFields || []);
+
+  return (
+    <ConnectionTypePreviewDrawer
+      isExpanded={isDrawerExpanded}
+      onClose={() => setIsDrawerExpanded(false)}
+      obj={createConnectionTypeObj(
+        {
+          k8sName: translateDisplayNameForK8s(connectionNameDesc.name),
+          displayName: connectionNameDesc.name,
+          description: connectionNameDesc.description,
+          enabled: connectionEnabled,
+          username: '',
+        },
+        connectionFields,
+      )}
+    >
+      <ApplicationsPage
+        title="Create connection type"
+        loaded
+        empty={false}
+        errorMessage="Unable load to connection types"
+        breadcrumb={<CreateConnectionTypeBreadcrumbs />}
+        headerAction={
+          <Button
+            variant="secondary"
+            icon={<OpenDrawerRightIcon />}
+            aria-expanded={isDrawerExpanded}
+            onClick={() => setIsDrawerExpanded(!isDrawerExpanded)}
+            data-testid="preview-drawer-toggle-button"
+          >
+            Preview
+          </Button>
+        }
+      >
+        <PageSection isFilled variant="light">
```

Comment:
would be better to tighten it up by removing the top padding on this section
```suggestion
        <PageSection className="pf-v5-u-pt-0 isFilled variant="light">
```